### DISCUSSION TITLE
[js] Should multiple FFI imports compile to a single ESM re-export?

### DISCUSSION THREAD

**Discussion by hayleigh-dot-dev** [gleam-lang/gleam#1168] [Ideas & suggestions]
Currently when compiling a Gleam module with multiple external fns from the same source, we compile to multiple import/export pairs:
```js
import { from } from "./ref.ffi.js";
export { from };

import { get } from "./ref.ffi.js";
export { get };

import { set } from "./ref.ffi.js";
export { set };

import { update } from "./ref.ffi.js";
export { update };
```
This can actually be reduced to just one line using [ESM re-exports](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#re-exporting_aggregating), resulting in:
```js
export { from, get, set, update } from "./ref.ffi.js";
```

I'm not sure whether this is a tangible win or not, I believe imports are cached so it's not like you'd hit the network multiple times with the current approach and it's nice to have the compiled module more-or-less follow the order of declaration that the original source did.

**Comment by lpil** [comment]
Yes, definitely. This makes much nicer looking code 
https://github.com/gleam-lang/gleam/issues/1170

**Comment by lpil** [comment]
Locking this as it is implemented! Thank you

**Reply by lpil** [reply]
Hmm that did not do what I expected it to do.
